octobot_trading/data/position.py

```diff
@@ -57,37 +67,52 @@ def _update(self, position_id, symbol, currency, market, timestamp, entry_price,
                 self.creation_time = self.exchange_manager.exchange.get_uniform_timestamp(timestamp)
             self.timestamp = self.creation_time
 
-        if quantity and self.quantity != quantity:
-            self.quantity = quantity
+        if self._should_change(self.quantity, quantity):
+            self.quantity = float(quantity)
+            changed = True
+
+        if self._should_change(self.value, value):
+            self.value = float(value)
+            changed = True
+
+        if self._should_change(self.margin, margin):
+            self.margin = float(margin)
+            changed = True
+
+        if self._should_change(self.unrealised_pnl, unrealised_pnl):
+            self.unrealised_pnl = float(unrealised_pnl)
             changed = True
 
-        if unrealised_pnl and self.unrealised_pnl != unrealised_pnl:
-            self.unrealised_pnl = unrealised_pnl
+        if self._should_change(self.realised_pnl, realised_pnl):
+            self.realised_pnl = float(realised_pnl)
             changed = True
 
-        if leverage and self.leverage != leverage:
-            self.leverage = leverage
+        if self._should_change(self.leverage, leverage):
+            self.leverage = int(leverage)
             changed = True
 
-        if entry_price and self.entry_price != entry_price:
-            self.entry_price = entry_price
+        if self._should_change(self.entry_price, entry_price):
+            self.entry_price = float(entry_price)
 
-        if mark_price and self.mark_price != mark_price:
-            self.mark_price = mark_price
+        if self._should_change(self.mark_price, mark_price):
+            self.mark_price = float(mark_price)
 
-        if liquidation_price and self.liquidation_price != liquidation_price:
-            self.liquidation_price = liquidation_price
+        if self._should_change(self.liquidation_price, liquidation_price):
+            self.liquidation_price = float(liquidation_price)
 
-        if status and self.status != status:
-            self.status = status
+        if self._should_change(self.status.value, status):
+            self.status = PositionStatus(status)
 
-        if side and self.side != side:
-            self.side = side
+        if self._should_change(self.side.value, side):
+            self.side = PositionSide(side)
+
+        if self.side is PositionSide.UNKNOWN and self.quantity:
+            self.side = PositionSide.LONG if self.quantity > 0 else PositionSide.SHORT
 
         return changed
 
     def is_liquidated(self):
-        return self.status is PositionStatus.LIQUIDATED
+        return self.status is PositionStatus.LIQUIDATING
```

Comment:
When is a position ```LIQUIDATED```  ?

Reply:
Actually a position is never liquidated. It can be in liquidation aka `LIQUIDATING`. When the position quantity has been liquidated the position is closed.

Reply:
Good to know !